File: request-manager/src/main/java/fii/request/manager/domain/Workflow.java
package fii.request.manager.domain;

import jakarta.persistence.*;
import lombok.Data;
import java.util.Set;

@Entity
@Data
public class Workflow {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;
    private String description;

    //LocalDateTime createdAt;

    @OneToMany(mappedBy = "workflow", cascade = CascadeType.ALL)
    private Set<WorkflowStep> workflowSteps;

    @OneToOne
    @JoinColumn(name = "workflow_sharing_id")
    private WorkflowSharing sharing;
}

File: workflow-manager-frontend/app/src/main/java/com/example/workflow_manager_frontend/domain/Workflow.kt
package com.example.workflow_manager_frontend.domain

import android.os.Parcelable
import androidx.room.Entity
import androidx.room.PrimaryKey
import kotlinx.parcelize.Parcelize

@Entity
@Parcelize
data class Workflow(
    var description: String = "",
    @PrimaryKey val id: Int? = 0,
    var name: String = "",
    val sharing: Sharing? = null
) : Parcelable